pandas/core/ops.py

```diff
@@ -505,11 +509,20 @@ def _convert_to_array(self, values, name=None, other=None):
         inferred_type = lib.infer_dtype(values)
         if (inferred_type in ('datetime64', 'datetime', 'date', 'time') or
                 is_datetimetz(inferred_type)):
+
+            if ovalues is pd.NaT and name == '__sub__':
```

Comment:
what's wrong this this code that is existing? (obviously the dtype is incorrect if other is dt64 but otherwise this is fine)
```
            # if we have a other of timedelta, but use pd.NaT here we
            # we are in the wrong path
            if (supplied_dtype is None and other is not None and
                (other.dtype in ('timedelta64[ns]', 'datetime64[ns]')) and
                    isna(values).all()):
                values = np.empty(values.shape, dtype='timedelta64[ns]')
                values[:] = iNaT
```

Reply:
For starters the other.dtype condition excludes datetime64s with timezones.  More generally, this PR is specifically about scalar subtraction of pd.NaT and I would rather give that its own block than try to shoe-horn it into an existing block.

Reply:
simplify this like the above code

Reply:
OK.  But in my head I'm going to put sarcasm quotes around the word "simplify".

Reply:
same issue. you are repeating code. you only need to select the dtype if other.dtype == 'timedelta64[ns]'
but the construction is regardless

Reply:
What you're suggesting would catch `np.array([np.datetime64('NaT')])` which is unambiguously datetimelike, or consider `np.array([pd.NaT, datetime64('NaT')])`.  `_TimeOp` is a mess in part because it is not clear about what cases it is trying to handle where.  This PR is trying to fix a bug tied to one very specific case.  The fix should not catch any cases other than that specific intended case.